Source/WebKit/NetworkProcess/storage/FileSystemStorageHandle.cpp

```diff
@@ -267,4 +268,51 @@ std::optional<FileSystemStorageError> FileSystemStorageHandle::move(WebCore::Fil
     return std::nullopt;
 }
 
+std::optional<WebCore::FileSystemSyncAccessHandleIdentifier> FileSystemStorageHandle::activeSyncAccessHandle()
+{
+    if (!m_activeSyncAccessHandle)
+        return std::nullopt;
+
+    return m_activeSyncAccessHandle->identifier;
+}
+
+bool FileSystemStorageHandle::isActiveSyncAccessHandle(WebCore::FileSystemSyncAccessHandleIdentifier accessHandleIdentifier)
+{
+    return m_activeSyncAccessHandle && m_activeSyncAccessHandle->identifier == accessHandleIdentifier;
+}
+
+uint64_t FileSystemStorageHandle::allocatedUnusedCapacity()
+{
+    if (!m_activeSyncAccessHandle)
+        return 0;
+
+    auto actualSize = valueOrDefault(FileSystem::fileSize(m_path));
+    return actualSize > m_activeSyncAccessHandle->capacity ? 0 : m_activeSyncAccessHandle->capacity - actualSize;
+}
+
+void FileSystemStorageHandle::requestNewCapacityForSyncAccessHandle(WebCore::FileSystemSyncAccessHandleIdentifier accessHandleIdentifier, uint64_t newCapacity, CompletionHandler<void(std::optional<uint64_t>)>&& completionHandler)
+{
+    if (!isActiveSyncAccessHandle(accessHandleIdentifier))
+        return completionHandler(std::nullopt);
+
+    if (newCapacity <= m_activeSyncAccessHandle->capacity)
+        return completionHandler(m_activeSyncAccessHandle->capacity);
+
+    if (!m_manager)
+        return completionHandler(std::nullopt);
+
+    uint64_t spaceRequested = newCapacity - m_activeSyncAccessHandle->capacity;
```

Comment:
WebProcess asks exactly what it needs and no more.
We should probably request more than the exact space requested, either from WebProcess or even better here in NetworkProcess.

Reply:
Yes, I will make a follow-up patch to request more.